projects/slap_manipulation/src/slap_manipulation/env/language_planner_env.py

```diff
@@ -0,0 +1,250 @@
+from typing import Any, Dict, List, Optional
+
+import numpy as np
+import rospy
+
+import home_robot
+from home_robot.core.interfaces import (
+    Action,
+    DiscreteNavigationAction,
+    HybridAction,
+    Observations,
+)
+from home_robot.utils.geometry import xyt2sophus
+from home_robot_hw.env.stretch_pick_and_place_env import DETIC, StretchPickandPlaceEnv
+
+
+class LanguagePlannerEnv(StretchPickandPlaceEnv):
+    def __init__(
+        self,
+        config,
+        goal_options: List[str] = None,
+        segmentation_method: str = DETIC,
+        visualize_planner: bool = False,
+        ros_grasping: bool = True,
+        test_grasping: bool = False,
+        dry_run: bool = False,
+        debug: bool = False,
+        *args,
+        **kwargs
+    ):
+        super().__init__(
+            config,
+            goal_options,
+            segmentation_method,
+            visualize_planner,
+            ros_grasping,
+            test_grasping,
+            dry_run,
+            debug,
+            *args,
+            **kwargs
+        )
+        self.current_goal_id = None
+        self.current_goal_name = None
+
+    def reset(self):
+        # TODO (@priyam): clean goal info if in obs
+        rospy.sleep(0.5)  # Make sure we have time to get ROS messages
+        self.robot.wait()
+        self._episode_start_pose = xyt2sophus(self.robot.nav.get_base_pose())
+        if self.visualizer is not None:
+            self.visualizer.reset()
+
+        # Switch control mode on the robot to nav
+        # Also set the robot's head into "navigation" mode - facing forward
+        self.robot.move_to_nav_posture()
+
+    def set_goal(self, info: Dict):
+        vocab = ["other"] + info["object_list"] + ["other"]
+        self.segmentation.reset_vocab(vocab)
+
+        if len(info["object_list"]) > 1:
+            self.current_goal_id = 2
+            self.current_goal_name = info["object_list"][1]
+        else:
+            self.current_goal_id = 1
+            self.current_goal_name = info["object_list"][0]
+
+    def _switch_to_manip_mode(self, grasp_only=False):
```

Comment:
Why is this duplicated?

Reply:
A lot of differences in behavior. I didn't want to cause regressions in OVMM code with my changes. I am doing away with any notion of knowing `start_recep`, `object` and `goal_recep` at the begining.